### ISSUE TITLE
[no-magic-numbers] False positive with array/tuple type index access

### ISSUE THREAD

**Issue by davecardwell** [typescript-eslint/typescript-eslint#4780]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-magic-numbers": [
      "error",
      { "ignoreArrayIndexes": true }
    ],
  }
}
```

```TS
type Foo = ["bar", "baz"];
function foo(bar: string, baz: number) {}

// No magic number: 1. eslint(@typescript-eslint/no-magic-numbers)
type Oof1 = Foo[1];

// No magic number: 1. eslint(@typescript-eslint/no-magic-numbers)
type Oof2 = Parameters<typeof foo>[1];
```

**Expected Result**

I expect the array index access to be ignored.

**Actual Result**

> No magic number: 1. eslint(@typescript-eslint/no-magic-numbers)

**Additional Info**

An alternative would be to add an additional option for this if you don’t think it should be covered under “ignoreArrayIndexes”.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.17.0` |
| `@typescript-eslint/parser`        | `5.17.0` |
| `TypeScript`                       | `4.6.3` |
| `ESLint`                           | `8.12.0` |
| `node`                             | `17.8.0` |

**Comment by JoshuaKGoldberg** [comment]
Yeah, I think this makes sense as an opt-out option: ignoring indexes on tuple types such as those from `Parameters`. Maybe,`ignoreTypeIndexes`?

**Comment by davecardwell** [comment]
Thanks @JoshuaKGoldberg. I’ll take a look at creating a PR to add an `ignoreTypeIndexes` option.

**Comment by timdeschryver** [comment]
👋 @davecardwell , if you want, I can create a PR for this. 
I managed to get this working earlier today but I didn't see your comment until now.
Don't feel obligated though, if you want to contribute that's totally fine.

**Comment by davecardwell** [comment]
👋 @timdeschryver Whoops! I had also done the work after leaving my comment. I finished up the docs change just now and submitted #4789, but I haven’t worked with `estree` before so would be happy to close that PR in favor of yours if you recommend your approach instead.
